.devcontainer/Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM mcr.microsoft.com/devcontainers/go:1-1.21-bookworm
+
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends postgresql-client
+
+RUN curl -LO https://github.com/redpanda-data/redpanda/releases/download/v23.2.7/rpk-linux-amd64.zip \
+    && unzip rpk-linux-amd64.zip \
```

Comment:
Are we not going to validate the checksum here script kiddie?

Reply:
Added checksum validation, scripting this is new to me so let me know if I missed the mark.

Reply:
LGTM